packages/flex-plugin-scripts/src/config/compiler.ts

```diff
@@ -98,39 +104,62 @@ export default (config: Configuration, devServer: boolean, type = WebpackType.Co
         compiler.hooks.tsCompiled.call(messages.warnings, messages.errors);
       }
 
-      const isSuccessful = result.errors.length === 0 && result.warnings.length === 0;
-      if (isSuccessful) {
-        if (type === WebpackType.Static || type === WebpackType.Complete) {
-          devServerSuccessful(local, network, localPlugins);
-        }
-        return;
-      }
-
-      const formatted = webpackFormatMessages({
-        errors: result.errors,
-        warnings: result.warnings,
-      });
-
-      // Only show errors if both exist
-      if (stats.hasErrors()) {
-        // Most errors are duplicate of the same error
-        // So only show the first error
-        formatted.errors.length = 1;
-
-        logger.error(`Failed to compile plugin ${logger.colors.red.bold(getPaths().app.name)}.`);
-        logger.info(formatted.errors.join('\n'));
-        logger.newline();
-        return;
-      }
-
-      // Show warning messages
-      logger.warning(`Compiled plugin ${logger.colors.yellow.bold(getPaths().app.name)} with warning(s).`);
-      logger.info(formatted.warnings.join('\n'));
-      logger.newline();
+      onCompile({result, appName: getPaths().app.name});
     });
 
     return compiler;
   } catch (err) {
     throw new FlexPluginError('Failed to create a webpack compiler: ' + err.message);
   }
 };
+
+/**
+ * Prints the errors and warnings or a successful message when compilation finishes
+ * @param port    the port the server is running on
+ * @param plugins the local plugins running
+ * @param showSuccessMsg    whether to show succecss message or not
+ */
+export const onCompileComplete = (port: number, plugins: string[], showSuccessMsg: boolean): OnCompile => {
+  const { local, network } = getLocalAndNetworkUrls(port);
+  return ({ result, appName}) => {
+    logger.clearTerminal();
+    results[appName] = result;
```

Comment:
Need to keep track of ALL results so if multiple plugins have failed, and one is then fixed, we still show the errors for the other ones.